### PR TITLE
Optimize Contact Query used in Household Naming

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 1]
```diff
@@ -347,10 +347,78 @@ public class HH_HouseholdNaming_TEST {
...
+        insert accounts;
+
+        List<npo02__Household__c> hhs = buildHouseholds(2);
+        insert hhs;
+
+        List<Contact> contacts = new List<Contact>{
+            new Contact(FirstName = 'Alpha', LastName = 'A', AccountId = accounts[0].Id, npo02__Household__c = hhs[0].Id),
+            new Contact(FirstName = 'Beta', LastName = 'B', npo02__Household__c = hhs[1].Id),
+            new Contact(FirstName = 'Charlie', LastName = 'C', AccountId = accounts[1].Id),
+            new Contact(FirstName = 'Deleta', LastName = 'D', npe01__Private__c = true)
```
Comment (joy-sfdo) [inline src/classes/HH_HouseholdNaming_TEST.cls]: Typo -- Delta

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 2]
```diff
@@ -559,4 +627,170 @@ public class HH_HouseholdNaming_TEST {
...
+
+        hhIds = new List<Id>();
+        contacts = hhNaming.getHouseholdContacts(hhIds);
+        System.assertEquals(0, contacts.size(), 'The Contact list should contain no record for empty Id list arg: ' + contacts);
+        
+        hhIds.add(null);
+        contacts = hhNaming.getHouseholdContacts(hhIds);
+        System.assertEquals(0, contacts.size(), 'The Contact list should contain no record when Id list arg contains null: ' + contacts);
+
+        Id mockAccountId = Account.sObjectType.getDescribe().getKeyPrefix() + '000000000001AAA';
```
Comment (joy-sfdo) [inline src/classes/HH_HouseholdNaming_TEST.cls]: Create the mockAccountId as a constant.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 3]
```diff
@@ -347,10 +347,78 @@ public class HH_HouseholdNaming_TEST {
...
+        ));
+
+        List<Account> accounts = buildHHAccounts(2);
+        insert accounts;
+
+        List<npo02__Household__c> hhs = buildHouseholds(2);
+        insert hhs;
+
+        List<Contact> contacts = new List<Contact>{
+            new Contact(FirstName = 'Alpha', LastName = 'A', AccountId = accounts[0].Id, npo02__Household__c = hhs[0].Id),
```
Comment (joy-sfdo) [inline src/classes/HH_HouseholdNaming_TEST.cls]: Are both AccountId and npo02__Household__c needed?

File: src/classes/HH_HouseholdNaming_TEST.cls (hunk 2 above)
Comment (RadSFDeveloper) [reply src/classes/HH_HouseholdNaming_TEST.cls]: A mock Account Id has been created in `UTIL_UnitTestData_TEST` and used throughout unit tests.

File: src/classes/HH_HouseholdNaming_TEST.cls (hunk 3 above)
Comment (RadSFDeveloper) [reply src/classes/HH_HouseholdNaming_TEST.cls]: The Households are not required in this test; they have been removed.

**Pull request merged**